pkg/v3/types/interfaces.go

```diff
@@ -2,101 +2,77 @@ package types
 
 import (
 	"context"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/types/automation"
 )
 
-type UpkeepTypeGetter func(UpkeepIdentifier) UpkeepType
-type WorkIDGenerator func(UpkeepIdentifier, Trigger) string
+//go:generate mockery --name Encoder --structname MockEncoder --srcpkg "github.com/smartcontractkit/chainlink-common/pkg/types/automation" --case underscore --filename encoder.generated.go
```

Comment:
Do we actually need to generate mocks here? can't we use generated mocks from cll-common?

Reply:
Initially these mocks were migrated to common, but other developers weren't keen on having the mocks in common, as they weren't needed directly in common. So instead, the mocks are being generated in automation, since automation is the only repo that needs them